packages/subcommands/src/lib/Subcommand.ts

```diff
@@ -0,0 +1,244 @@
+import { Command, Result, UserError, type Args, type ChatInputCommand, type MessageCommand, type PieceContext } from '@sapphire/framework';
+import type { CacheType, Message } from 'discord.js';
+import type {
+	ChatInputCommandSubcommandMappingMethod,
+	MessageSubcommandMappingMethod,
+	SubcommandMappingArray,
+	SubcommandMappingMethod
+} from './SubcommandMappings';
+import {
+	SubcommandPluginEvents,
+	SubcommandPluginIdentifiers,
+	type ChatInputSubcommandAcceptedPayload,
+	type MessageSubcommandAcceptedPayload
+} from './types/Events';
+
+export class Subcommand<PreParseReturn extends Args = Args, O extends Subcommand.Options = Subcommand.Options> extends Command<PreParseReturn, O> {
+	public parsedSubcommandMappings: SubcommandMappingArray;
+
+	public constructor(context: PieceContext, options: O) {
+		super(context, options);
+		this.parsedSubcommandMappings = options.subcommands ?? [];
+	}
+
+	public onLoad() {
+		super.onLoad();
+
+		const externalMapping: SubcommandMappingArray | undefined = Reflect.get(this, 'subcommandMappings');
```

Comment:
Because I am using fields for it (could move to options but I like fields), could we export a separate class, ala AllFlowsPrecondition, that makes this field abstract? 🥺 Its not a PR blocker but a nice to have

Reply:
I don't follow. Can you dumb this down for me?

Reply:
it's exported as a type so you just type your field as this... 
```ts
private readonly subcommandMappings: SubcommandMappingArray = [

]
```